source/Engine.cpp

```diff
@@ -319,20 +319,19 @@ void Engine::Place(const list<NPC> &npcs, shared_ptr<Ship> flagship)
 {
 	for(const NPC &npc : npcs)
 	{
-		map<Ship *, int> droneCarriers;
-		map<Ship *, int> fighterCarriers;
+		set<Ship *> carriers;
```

Comment:
This could be inside a map keyed by ship category to make handling of large numbers of NPC carried ships more intelligent.  (Could also keep the `map<Ship *, int>` signature too instead of using a set.)

Reply:
Yes, changed the code to use a map of maps (outer map with the category, inner map with the signature as given/original).